charleslong13/Elite-Vixen-Paranormal-Capstone-Project
# csl-initialCommit

#### Changes Made
1. Added all currently available files with this initial commit 
Applies to issue(s): #10 #7 #8 #11 
​Created all current repositories and files. Majority of baseline functionality is included plus some CSS. Managed to actually get some of my stretch goals done earlier than expected, but still need to work on crucial components for my MVP. 
Login page is created and functions, but needs adjustment and is not completed. 
Homepage and post feed are functional. 
Create new post button functionality is working. 
Favoriting/unfavoriting posts is functional (may need adjusting)
User profiles and direct messaging are functional. 


#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout csl-initialBranch
    ```
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
3. To test app functionality:

    > Starting the API: Open a terminal window and go to the Elite-Vixen-Paranormal-Capstone-Project directory.
Then enter:
cd api 
json-server database.json -p 8088 -w

Starting the Web Server Open another terminal window and go to the Elite-Vixen-Paranormal-Capstone-Project directory.
Then enter:
cd src 
serve

Login page is created and functions, but needs adjustment and is not completed. Currently the user should still be able to either log in with an existing account, or create a new account with the current form and be able to log in.

Once logged in the user should be taken to the Homepage where they should see the nav bar with links to different page views at the top(they are all there, but not all functional yet), you should see a home icon, About, Contact Us, Investigation Request, message icon, new message counter icon, and sign out buttons/links currently displayed 

Three of the navbar links are currently working and can be tested: 
- The messages icon when clicked should render a new message form that when submitted, posts a new message to the messages array in the database 
- The new message counter icon should dynamically display the number of unread messages for the currently logged in user, when clicked it should render a message list with all of that user's currently unread messages, once viewed the messages "read" property should be set to true and the number counter should be reset 

Below the navbar the user should see a button to create a new post, the post feed- posts displayed in a list centered in the page, and a footer with options for filtering (currently exists but needs modification) 

- Test that the Create new post button functionality is working by viewing the networks tab and submitting a new post.
- Test that the Favoriting/unfavoriting posts is functional (may need adjusting)
- Test that the direct messaging is functional by clicking the messaging icon to fill out the form (make sure the dropdown is rendering all of the users correctly), select yourself as the recipient, then click the send button while viewing the network tab to ensure the message object is getting posted to the messages array in the database with the proper keys. When the page re-renders, the message counter icon's number should have increased by 1. When clicked, it should display the message you just sent yourself. If you navigate back to the home page, the message counter should be reset to 0. 
- Test clicking a user's name on a post to see if the User's profile information gets rendered properly. You should see the clicked on user's name, their total number of submitted posts, and then messages between the clicked on user and the currently logged in user. 


4. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.
